internal/api/option/auth.go

```diff
@@ -410,29 +418,40 @@ func (a *authInterceptor) authenticate(
 	), nil
 }
 
+var verifierMu = sync.Mutex{}
+
 // verifyIDPIssuedToken attempts to verify that the provided raw token was
 // issued by Kargo's OpenID Connect identity provider. On success, select claims
-// are extracted and returned along with a true boolean. If the provided raw
-// token couldn't be verified, the returned boolean is false. A non-nil error is
-// only ever returned if something goes wrong AFTER successfully verifying the
-// token. Callers may infer that if the returned error is nil, but the returned
-// boolean is false, the provided raw token could not be verified.
+// are extracted and returned.
 func (a *authInterceptor) verifyIDPIssuedToken(
 	ctx context.Context,
 	rawToken string,
-) (claims, bool) {
+) (claims, error) {
+	if a.cfg.OIDCConfig == nil {
+		// Really, this method never should have been called under these
+		// circumstances.
+		return claims{}, errors.New("OpenID Connect is not supported")
+	}
 	c := claims{}
 	if a.oidcTokenVerifyFn == nil {
-		return c, false
+		verifierMu.Lock()
+		defer verifierMu.Unlock()
```

Comment:
We probably want to unlock the mutex as soon as we can, and there appears to be quite some logic here before the `defer` will be called.

Reply:
Initially I was and then opted for clarity. The claim extraction is very quick. They're just raw claims. It doesn't include any of the claims --> SA mapping logic.

Reply:
Ah... but there's also the actual verification before that, which is a cryptographic process which may be a tad slow.

I'll fix.

Reply:
Fixed in 445cdfe72421ecc83d3eee2d7218cc0ce034f83b